### PR TITLE
[CELEBORN-1276] Move checkQuotaSpaceAvailable from Quota to QuotaManager

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?
Move checkQuotaSpaceAvailable from Quota to QuotaManager

### Why are the changes needed?
Put method in correct place

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Added UT

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
ping @RexXiong

**Review by RexXiong — APPROVED** [review]
LGTM, thanks!

**Comment by AngersZhuuuu** [comment]
Thanks, merge to main(v0.5.0)

File: master/src/main/scala/org/apache/celeborn/service/deploy/master/quota/QuotaManager.scala [hunk 1]
```diff
@@ -37,4 +38,74 @@ class QuotaManager(celebornConf: CelebornConf, configService: ConfigService) ext
...
+      checkDiskBytesWritten(userIdentifier, resourceResumption.diskBytesWritten, quota),
+      checkDiskFileCount(userIdentifier, resourceResumption.diskFileCount, quota),
+      checkHdfsBytesWritten(userIdentifier, resourceResumption.hdfsBytesWritten, quota),
+      checkHdfsFileCount(userIdentifier, resourceResumption.hdfsFileCount, quota))
+    val exceed = checkResults.foldLeft(false)(_ || _._1)
+    val reason = checkResults.foldLeft("")(_ + _._2)
+    (!exceed, reason)
+  }
+
+  private def checkDiskBytesWritten(
```
Comment (SteNicholas) [inline master/src/main/scala/org/apache/celeborn/service/deploy/master/quota/QuotaManager.scala:56]: Could the check methods merge into one method? Most implementation of check methods are similar.

Comment (AngersZhuuuu) [reply master/src/main/scala/org/apache/celeborn/service/deploy/master/quota/QuotaManager.scala:56]: > Could the check methods merge into one method? Most implementation of check methods are similar.

Let me do a final refactor pr for all code related to quota management.

**Pull request closed without merging**